src/Functions/decodeXMLComponent.cpp

```diff
@@ -0,0 +1,249 @@
+#include <Columns/ColumnString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionStringToString.h>
+#include <common/find_symbols.h>
+
+#include <cstdio>
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+namespace
+{
+    struct DecodeXMLComponentName
+    {
+        static constexpr auto name = "decodeXMLComponent";
+    };
+
+    class FunctionDecodeXMLComponentImpl
+    {
+    public:
+        static void vector(
+            const ColumnString::Chars & data,
+            const ColumnString::Offsets & offsets,
+            ColumnString::Chars & res_data,
+            ColumnString::Offsets & res_offsets)
+        {
+            res_data.resize(data.size());
+            size_t size = offsets.size();
+            res_offsets.resize(size);
+
+            size_t prev_offset = 0;
+            size_t res_offset = 0;
+
+            for (size_t i = 0; i < size; ++i)
+            {
+                const char * src_data = reinterpret_cast<const char *>(&data[prev_offset]);
+                size_t src_size = offsets[i] - prev_offset;
+                size_t dst_size = execute(src_data, src_size, reinterpret_cast<char *>(res_data.data() + res_offset));
+
+                res_offset += dst_size;
+                res_offsets[i] = res_offset;
+                prev_offset = offsets[i];
+            }
+
+            res_data.resize(res_offset);
+        }
+
+        [[noreturn]] static void vectorFixed(const ColumnString::Chars &, size_t, ColumnString::Chars &)
+        {
+            throw Exception("Function decodeXMLComponent cannot work with FixedString argument", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+
+    private:
+        static const int min_XML_number = 32;
+        static const int max_XML_number = 126;
```

Comment:
I thought it should support all unicode.

Reply:
Characters beyond this range cannot be printed.

Reply:
Please open developer tools in the browser (press F12), open "console" and run the following snippet:

`document.body.innerHTML = "&#20026;&#20160;&#20040;&#19981;"`

Reply:
Thanks for your advice, I have added the support of decode Unicode.